src/argilla/client/feedback.py

```diff
@@ -0,0 +1,454 @@
+        if len(self.__records) < 1:
+            raise RuntimeError(
+                "In order to get items from `rg.FeedbackDataset` you need to either add them first with `add_records` or fetch them from Argilla or HuggingFace with `fetch_records`."
+            )
+        if isinstance(key, int) and len(self.__records) < key:
+            raise IndexError(f"This dataset contains {len(self)} records, so index {key} is out of range.")
+        return self.__records[key]
+
+    def __del__(self) -> None:
+        if hasattr(self, "__records"):
+            del self.__records
+
+    def __enter__(self) -> "FeedbackDataset":
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        self.__del__()
+
+    @property
+    def guidelines(self) -> str:
+        return self.__guidelines
+
+    @guidelines.setter
+    def guidelines(self, guidelines: str) -> None:
+        if not isinstance(guidelines, str):
+            raise TypeError(f"Expected `guidelines` to be a string, got {type(guidelines)} instead.")
+        self.__guidelines = guidelines
+
+    @property
+    def fields(self) -> List[FeedbackFieldModel]:
+        return self.__fields
+
+    @property
+    def questions(self) -> List[FeedbackQuestionModel]:
+        return self.__questions
+
+    @property
+    def records(self) -> List["FeedbackItemModel"]:
+        return self.__records
+
+    def fetch_records(self, overwrite: bool = False) -> List["FeedbackItemModel"]:
+        if not self.argilla_id:
+            warnings.warn(
+                "No records have been logged into neither Argilla nor HuggingFace, so no records will be fetched. The current records will be returned instead."
+            )
+            return self.__records
+        if len(self.__records) > 0 and not overwrite:
+            warnings.warn(
+                "The current `rg.FeedbackDataset` contains records, just the pushed records will be fetched, while the non-pushed records will be lost. To ignore this warning and fetch the pushed records use `overwrite=True`. The current records will be returned instead."
+            )
+            return self.__records
+
+        if self.argilla_id:
+            client = rg.active_client()
+            first_batch = client.get_records(id=self.argilla_id, offset=0, limit=FETCHING_BATCH_SIZE)
+            self.__records = first_batch.items
+            total_batches = first_batch.total // FETCHING_BATCH_SIZE
+            current_batch = 1
+            with tqdm(initial=current_batch, total=total_batches, desc="Fetching records from Argilla") as pbar:
+                while current_batch <= total_batches:
+                    batch = client.get_records(
+                        id=self.argilla_id, offset=FETCHING_BATCH_SIZE * current_batch, limit=FETCHING_BATCH_SIZE
+                    )
+                    self.__records += batch.items
```

Comment:
If we do:

```python
ds.fetch_records(override=True)
```
and later 
```
ds.fetch_records(override=False) # This will show a message that can be confusing regarding the non-pushed records (which are records fetched previously)